pages/date-range-picker/simple.page.tsx

```diff
@@ -13,15 +13,40 @@ import {
   SpaceBetween,
 } from '~components';
 
-import { i18nStrings, i18nStringsDateOnly, isValid, relativeOptions } from './common';
+import AppContext from '../app/app-context';
+import {
+  applyDisabledReason,
+  checkIfDisabled,
+  DateRangePickerDemoContext,
+  dateRangePickerDemoDefaults,
+  DisabledDate,
+  generateI18nStrings,
+  generatePlaceholder,
+  generateRelativeOptions,
+  isValid,
+} from './common';
 
 export default function DatePickerScenario() {
-  const [showRelativeOptions, setShowRelativeOptions] = useState(true);
-  const [dateOnly, setDateOnly] = useState(false);
-  const [invalid, setInvalid] = useState(false);
-  const [warning, setWarning] = useState(false);
-  const [rangeSelectorMode, setRangeSelectorMode] = useState<DateRangePickerProps.RangeSelectorMode>('default');
-  const [value, setValue] = useState<DateRangePickerProps['value']>(null);
+  const { urlParams, setUrlParams } = useContext(AppContext as DateRangePickerDemoContext);
+  const [value, setValue] = useState<DateRangePickerProps['value']>({
+    type: 'absolute',
+    startDate: '2024-12-30T00:00:00+01:00',
+    endDate: '2024-12-31T23:59:59+01:00',
+  });
+
+  const monthOnly = false;
+  const absoluteFormat =
+    urlParams.absoluteFormat ?? (dateRangePickerDemoDefaults.absoluteFormat as DateRangePickerProps.AbsoluteFormat);
+  const showRelativeOptions = urlParams.showRelativeOptions ?? dateRangePickerDemoDefaults.showRelativeOptions;
+  const dateOnly = urlParams.dateOnly ?? dateRangePickerDemoDefaults.dateOnly;
+  const disabledDates =
+    (urlParams.disabledDates as DisabledDate) ?? (dateRangePickerDemoDefaults.disabledDates as DisabledDate);
+  const withDisabledReason = urlParams.withDisabledReason ?? dateRangePickerDemoDefaults.withDisabledReason;
+  const invalid = urlParams.invalid ?? dateRangePickerDemoDefaults.invalid;
+  const warning = urlParams.warning ?? dateRangePickerDemoDefaults.warning;
+  const rangeSelectorMode =
+    urlParams.rangeSelectorMode ??
+    (dateRangePickerDemoDefaults.rangeSelectorMode as DateRangePickerProps.RangeSelectorMode);
```

Comment:
Wouldn't it be easier to make all defaults false/undefined, instead of writing this wall of text for every page?

Reply:
It's a tradeoff. I understand your approach. While it may seem verbose, using centralized defaults in dateRangePickerDemoDefaults allows us to maintain consistency across multiple pages and easily update default values in one place. This approach reduces the risk of inconsistencies and makes maintenance easier in the long run. It's worth noting that specific initial values can be assigned in place of the defaults if desired for any particular page. Also, not all pages have this 'wall of text,' as many don't allow for changes to as many props, resulting in a more concise implementation on those pages.

Reply:
Other integ tests screenshots added in CR-165057689